packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -214,10 +217,21 @@ export class ForkChoice implements IForkChoice {
           root: fromHexString(headRoot),
         });
       }
-
-      return (this.head = toBlockSummary(headNode));
+      prevHead = this.head;
+      return (this.head = newHead = toBlockSummary(headNode));
     } finally {
       if (timer) timer();
+      if (newHead && this.metrics) {
+        //if there is no prevHead, then isSameHead is false and isSameChain is true
+        const isSameHead = prevHead && toHexString(newHead.blockRoot) == toHexString(prevHead.blockRoot);
```

Comment:
If you need to compare byte arrays there's no need to convert them to strings, use byteArrayEquals